### PR TITLE
Store content package files in separate folders (WBL-62)

### PR DESCRIPTION
**Pull request by rakeshqdsa**
The content package should look like this:

/content_package
  /audio
  /images
  /items
  /passages
  /video
  imsmanifest.xml
The audio folder should contain any mp3 or other audio files.

The images folder should contain any jpg, gif, png etc files.

The items folder contains all XML assessment item files.

The passages folder contains all HTML passages.

The video folder should contain any mp4 or other video files.

https://learnosity.atlassian.net/browse/WBL-62

### CODE REVIEW

**Comment by michaelsharman** [comment]
The separate folders look good, but I didn't see any item tags converted as LOM in the QTI. This might be handled in another pull request, I'm not sure.

Also, this failed for me when importing into the TAO demo site.

**Comment by rakeshqdsa** [comment]
Both the thing ie conversion of item's tag into LOM in the manifest and importing into TAO is working fine from my side . 
I have also attached the content json .
[input.zip](https://github.com/Learnosity/learnosity-qti/files/3440456/input.zip)

**Comment by michaelsharman** [comment]
This is the zip I'm using. 
[raw.zip](https://github.com/Learnosity/learnosity-qti/files/3444148/raw.zip)

**Comment by rakeshqdsa** [comment]
Hi Michael
Please provide the item json for testing you have attached converted xml package . However this is also importing in tao .

**Comment by michaelsharman** [comment]
It would be one of the subfolder in this zip.
[learnosity.zip](https://github.com/Learnosity/learnosity-qti/files/3459077/learnosity.zip)

File: src/Services/ConvertToQtiService.php [hunk 1]
```diff
@@ -502,8 +557,21 @@ private function addAdditionalFileInfo($filesInfo, $files)
     {
         $files = array();
         foreach ($filesInfo as $info) {
+            $fileName = substr($info, strlen('/vendor/learnosity/itembank/assets/'));
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Services/ConvertToQtiService.php]: Done

File: src/Services/LearnosityToQtiPreProcessingService.php [hunk 2]
```diff
@@ -103,4 +109,23 @@ private function getFeatureReferenceFromClassName($classname)
         // TODO: throw exception
         return null;
     }
+
+    private function getSourceBasedOnMediaFormat($src)
+    {
+        $fileName = substr($src, strlen('/vendor/learnosity/itembank/assets/'));
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Services/LearnosityToQtiPreProcessingService.php]: Done

File: src/Services/ConvertToQtiService.php [hunk 3]
```diff
@@ -502,8 +557,21 @@ private function addAdditionalFileInfo($filesInfo, $files)
...
+            $mimeType = MimeUtil::guessMimeType($fileName);
+            $mediaFormatArray = explode('/', $mimeType);
             $file = new File();
-            $href = substr($info, strlen('/vendor/learnosity/itembank/'));
+            $href = '';
+            if (is_array($mediaFormatArray) && !empty($mediaFormatArray[0])) {
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Services/ConvertToQtiService.php]: Done

File: src/Services/LearnosityToQtiPreProcessingService.php [hunk 4]
```diff
@@ -103,4 +109,23 @@ private function getFeatureReferenceFromClassName($classname)
...
+        $mimeType = MimeUtil::guessMimeType($fileName);
+        $mediaFormatArray = explode('/', $mimeType);
+        $href = '';
+        if (is_array($mediaFormatArray) && !empty($mediaFormatArray[0])) {
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Services/LearnosityToQtiPreProcessingService.php]: Done

File: src/Services/LearnosityToQtiPreProcessingService.php [hunk 5]
```diff
@@ -103,4 +109,23 @@ private function getFeatureReferenceFromClassName($classname)
         // TODO: throw exception
         return null;
     }
+
+    private function getSourceBasedOnMediaFormat($src)
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Services/LearnosityToQtiPreProcessingService.php]: Done

**Pull request merged**